src/main/groovy/org/pih/warehouse/inventory/CycleCountCustomItemCommand.groovy

```diff
@@ -0,0 +1,41 @@
+package org.pih.warehouse.inventory
+
+import grails.databinding.BindUsing
+import grails.validation.Validateable
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.product.Product
+import org.springframework.web.context.request.RequestContextHolder
+
+class CycleCountCustomItemCommand implements Validateable {
+
+    boolean recount
+
+    @BindUsing({ obj, source ->
+        Product product = Product.read(source['inventoryItem']['product'])
+        Date expirationDate = new Date(source['inventoryItem']['expirationDate'])
+        return InventoryItem.findByProductAndLotNumberAndExpirationDate(product, source['inventoryItem']['lotNumber'], expirationDate)
+    })
+    InventoryItem inventoryItem
+
+    Location binLocation
+
+    Integer quantityCounted
+
+    String comment
+
+    CycleCount cycleCount
+
+    Location facility
+
+    def beforeValidate() {
+        String cycleCountId = RequestContextHolder.getRequestAttributes().params?.cycleCountId
+        String facilityId = RequestContextHolder.getRequestAttributes().params?.facility
+        cycleCount = CycleCount.read(cycleCountId)
+        facility = Location.read(facilityId)
+    }
+
+    static constraints = {
+        quantityCounted(nullable: true) // FIXME: should we allow for empty quantityCounted?
```

Comment:
hmm I would say yes (we know we need to count this new bin but haven't done it yet) but maybe that's a question for Manon.

Reply:
I think the answer is yes from an API perspective. The fact that we currently create the cycle count items automatically from the cycle count service is good for the current UI, but there might be a need to create cycle count items from the client for use cases that aren't relatd to this "add missing inventory item to cycle count" use case.